app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,320 @@
+    }
+
+    /**
+     * Test updateMapAndList method updates parent fragment view with camera target location
+     * at search custom area mode
+     */
+    @Test
+    fun testPlacesPopulatedForCameraTargetLocationWhenSearchCustomArea() {
+        nearbyPresenter.setNearbyLocked(false)
+        whenever(nearbyParentFragmentView.isNetworkConnectionEstablished()).thenReturn(true)
+        whenever(nearbyParentFragmentView.getLastLocation()).thenReturn(latestLocation)
+        whenever(nearbyParentFragmentView.getCameraTarget()).thenReturn(cameraTarget)
+        nearbyPresenter.updateMapAndList(LocationServiceManager.LocationChangeType.SEARCH_CUSTOM_AREA)
+        verify(nearbyParentFragmentView).populatePlaces(cameraTarget)
+    }
+
+    /**
+     * Test testUpdateMapAndList tracks users location if current location marker is visible and
+     * location is slightly changed
+     */
+    @Test
+    fun testUserTrackedWhenCurrentLocationMarkerVisible() {
+        nearbyPresenter.setNearbyLocked(false)
+        whenever(nearbyParentFragmentView.isNetworkConnectionEstablished()).thenReturn(true)
+        whenever(nearbyParentFragmentView.getLastLocation()).thenReturn(latestLocation)
+        whenever(nearbyParentFragmentView.isCurrentLocationMarkerVisible()).thenReturn(true)
+        nearbyPresenter.updateMapAndList(LocationServiceManager.LocationChangeType.LOCATION_SLIGHTLY_CHANGED)
+        verify(nearbyParentFragmentView).recenterMap(latestLocation)
+    }
+
+    /**
+     * Test testUpdateMapAndList doesn't track users location if current location marker is
+     * invisible and location is slightly changed
+     */
+    @Test
+    fun testUserNotTrackedWhenCurrentLocationMarkerInvisible() {
+        nearbyPresenter.setNearbyLocked(false)
+        whenever(nearbyParentFragmentView.isNetworkConnectionEstablished()).thenReturn(true)
+        whenever(nearbyParentFragmentView.getLastLocation()).thenReturn(latestLocation)
+        whenever(nearbyParentFragmentView.isCurrentLocationMarkerVisible()).thenReturn(false)
+        nearbyPresenter.updateMapAndList(LocationServiceManager.LocationChangeType.LOCATION_SLIGHTLY_CHANGED)
+        verify(nearbyParentFragmentView).isNetworkConnectionEstablished()
+        verify(nearbyParentFragmentView).getLastLocation()
+        verify(nearbyParentFragmentView).isCurrentLocationMarkerVisible()
+        verifyNoMoreInteractions(nearbyParentFragmentView)
+    }
+
+    /**
+     * Test search this area button became visible after user moved the camera target to far
+     * away from current target. Distance between these two point is 111.19 km, so our camera target
+     * is at outside of previously searched region if we set latestSearchRadius below 111.19. Thus,
+     * setSearchThisAreaButtonVisibility(true) should be verified.
+     */
+    @Test
+    fun testSearchThisAreaButtonVisibleWhenMoveToFarPosition() {
+        NearbyController.latestSearchLocation = Mockito.spy(LatLng(2.0,1.0,0.0F))
```

Comment:
why are they spied? mocks or just the objects themselves would do I think

Reply:
Solved

Reply:
They are still spied?